frontend/app/ui/UploadForm/UploadForm.tsx

```diff
@@ -64,14 +66,17 @@ export default function UploadForm() {
               list={[{ id: 1, value: "Full stack Developer" }]}
               onChange={handleOnChange}
             />
-            <FileUpload onChange={handleFileChange}/>
+            <FileUpload onChange={handleFileChange} />
           </div>
           <Button
             variant="contained"
             sx={{ width: "300px", margin: "8px", background: "#2fbea9" }}
             onClick={async () => {
               var response = await uploadResume(formdata);
-              router.push("/Results");
+              if (response) {
+                dispatch(updateResults(response));
+                router.push("/Results");
```

Comment:
you can also redirect the user from upload resume server action (if implemented) and pass data to /Results route I think. Can you check for it's possibility?